pkg/controller/servicebindingrequest/reconciler.go

```diff
@@ -0,0 +1,527 @@
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+	logf "sigs.k8s.io/controller-runtime/pkg/runtime/log"
+
+	v1alpha1 "github.com/redhat-developer/service-binding-operator/pkg/apis/apps/v1alpha1"
+)
+
+const (
+	connectsToLabel                = "connects-to"
+	almDescriptorPrefix            = "urn:alm:descriptor"
+	almKubernetesPrefix            = "io.kubernetes"
+	almServiceBindingRequestPrefix = "io.servicebindingrequest"
+)
+
+// Reconciler reconciles a ServiceBindingRequest object
+type Reconciler struct {
+	client client.Client
+	scheme *runtime.Scheme
+}
+
+// decodeString encoded with base64.
+func decodeString(encoded string) (string, error) {
+	decoded, err := base64.StdEncoding.DecodeString(encoded)
+	if err != nil {
+		return "", err
+	}
+	return string(decoded), nil
+}
+
+// selectCRDDescription based on ServiceBindingRequest and a list of CSV (Cluster Service Version)
+// picking the one that matches backing-selector rule, and returning the CRD description object.
+func (r *Reconciler) selectCRDDescription(
+	logger logr.Logger,
+	instance *v1alpha1.ServiceBindingRequest,
+	csvList *olmv1alpha1.ClusterServiceVersionList,
+) *olmv1alpha1.CRDDescription {
+	// based on backing-selector, looking for custom resource definition
+	backingSelector := instance.Spec.BackingSelector
+
+	logger.WithValues(
+		"BackingSelector.ResourceName", backingSelector.ResourceName,
+		"BackingSelector.ResourceVersion", backingSelector.ResourceVersion,
+	).Info("Looking for a CSV based on backing-selector")
+
+	for _, csv := range csvList.Items {
+		logger.WithValues("ClusterServiceVersion.Name", csv.Name).Info("Inspecting CSV...")
+		for _, crd := range csv.Spec.CustomResourceDefinitions.Owned {
+			logger = logger.WithValues(
+				"CRD.Name", crd.Name,
+				"CRD.Version", crd.Version,
+				"CRD.Kind", crd.Kind)
+			logger.Info("Inspecting CRD...")
+
+			// skipping entries that don't match backing selector name and version
+			if backingSelector.ResourceName != crd.Name {
+				continue
+			}
+			if crd.Version != "" && backingSelector.ResourceVersion != crd.Version {
+				continue
+			}
+
+			logger.Info("CRD matches backing-selector!")
+			return &crd
+		}
+	}
+
+	return nil
+}
+
+// selectCRDByName based in a list of CRDs select the one with matching name.
+func (r *Reconciler) selectCRDByName(
+	list *unstructured.UnstructuredList,
+	name string,
+) *unstructured.Unstructured {
+	for _, unstructuredObj := range list.Items {
+		if name == unstructuredObj.GetName() {
+			return &unstructuredObj
+		}
+	}
+	return nil
+}
+
+// pathValue read value of a given "path" inside of informed custom resource definition instance.
+func (r *Reconciler) pathValue(crd *unstructured.Unstructured, path string) (string, error) {
+	object := crd.Object
+	status, exists := object["status"].(map[string]interface{})
+	if !exists {
+		return "", fmt.Errorf("unable to find 'status' field in object '%s'", crd.GetName())
+	}
+
+	value, exists := status[path]
```

Comment:
Sometimes the value can be as a nested struct. For example - https://github.com/operator-framework/community-operators/blob/f980b60e6862cccb3f7eaf8bc2d9baff23a847e0/community-operators/etcd/etcdoperator.v0.9.2.clusterserviceversion.yaml#L298
Corresponding source code - https://github.com/coreos/etcd-operator/blob/8347d27afa18b6c76d4a8bb85ad56a2e60927018/pkg/apis/etcd/v1beta2/restore_types.go#L65
(For this sprint maybe we can omit this, but this should be under consideration)

Reply:
@Avni-Sharma indeed, that's a limitation we need a better approach to overcome than it's currently implemented. I think we could bring a library to aid in the dot-notation to extract data from a generic `map[string]interface{}` recursively.